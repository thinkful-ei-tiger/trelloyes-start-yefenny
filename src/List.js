import React from 'react';
import './List.css';

import Card from './Card';

function List(props) {
  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {props.cards.map((card) => (
          <Card
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            deleteCard={props.deleteCard}
          />
        ))}
        <button
          type='button'
          className='List-add-button'
          onClick={() => props.addRandomCard(props.id)}
        >
          + Add Random Card
        </button>
      </div>
    </section>
  );
}

export default List;
