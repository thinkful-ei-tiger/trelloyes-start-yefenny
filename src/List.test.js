import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List cards={[]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('render the UI as expected', () => {
    const tree = renderer.create(<List cards={[]} header='List' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
