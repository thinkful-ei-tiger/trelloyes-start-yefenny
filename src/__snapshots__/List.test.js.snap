// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component render the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
